aptos_sdk/asymmetric_crypto_wrapper.py

```diff
@@ -85,3 +90,133 @@ def deserialize(deserializer: Deserializer) -> Signature:
+                    data, signature
+                ), "Unable to verify signature"
+
+        except Exception:
+            return False
+        return True
+
+    @staticmethod
+    def from_crypto_bytes(indata: bytes) -> MultiPublicKey:
+        deserializer = Deserializer(indata)
+        return deserializer.struct(MultiPublicKey)
+
+    def to_crypto_bytes(self) -> bytes:
+        serializer = Serializer()
+        serializer.struct(self)
+        return serializer.output()
+
+    @staticmethod
+    def deserialize(deserializer: Deserializer) -> MultiPublicKey:
+        keys = deserializer.sequence(PublicKey.deserialize)
+        threshold = deserializer.u8()
+        return MultiPublicKey(keys, threshold)
+
+    def serialize(self, serializer: Serializer):
+        serializer.sequence(self.keys, Serializer.struct)
+        serializer.u8(self.threshold)
+
+
+class MultiSignature(asymmetric_crypto.Signature):
+    signatures: List[Tuple[int, Signature]]
+    BITMAP_NUM_OF_BYTES: int = 4
+
+    def __init__(self, signatures: List[Tuple[int, asymmetric_crypto.Signature]]):
+        # Sort first to ensure no issues in order
+        # signatures.sort(key=lambda x: x[0])
+        self.signatures = []
+        for index, signature in signatures:
+            assert (
+                index < self.BITMAP_NUM_OF_BYTES * 8
+            ), "bitmap value exceeds maximum value"
+            if isinstance(signature, Signature):
+                self.signatures.append((index, signature))
+            else:
+                self.signatures.append((index, Signature(signature)))
+
+    def __eq__(self, other: object):
+        if not isinstance(other, MultiSignature):
+            return NotImplemented
+        return self.signatures == other.signatures
+
+    def __str__(self) -> str:
+        return f"{self.signatures}"
+
+    @staticmethod
+    def deserialize(deserializer: Deserializer) -> MultiSignature:
+        signatures = deserializer.sequence(Signature.deserialize)
+        deserializer.uleb128()
+        bitmap = deserializer.u32()
+        num_bits = MultiSignature.BITMAP_NUM_OF_BYTES * 8
+        sig_index = 0
+        indexed_signatures = []
+
+        for i in range(0, num_bits):
+            has_signature = (bitmap & index_to_bitmap_value(i)) != 0
+            if has_signature:
+                indexed_signatures.append((i, signatures[sig_index]))
+                sig_index += 1
+
+        return MultiSignature(signatures)
+
+    def serialize(self, serializer: Serializer):
+        actual_sigs = []
+        bitmap = 0
+
+        for i, signature in self.signatures:
+            bitmap |= index_to_bitmap_value(i)
+            actual_sigs.append(signature)
+
+        serializer.sequence(actual_sigs, Serializer.struct)
+        serializer.uleb128(self.BITMAP_NUM_OF_BYTES)
+        serializer.u32(bitmap)
+
+
+def index_to_bitmap_value(i: int) -> int:
+    bit = i % 8
+    byte = i.__floordiv__(8)
```

Comment:
```suggestion
    byte = i // 8
```
https://docs.python.org/3/library/operator.html#mapping-operators-to-functions